flink-runtime/src/main/java/org/apache/flink/runtime/state/RegisteredOperatorBackendStateMetaInfo.java

```diff
@@ -175,14 +176,19 @@ public boolean equals(Object obj) {
 				return false;
 			}
 
+			if (!(obj instanceof Snapshot)) {
+				return false;
+			}
+
+			Snapshot snapshot = (Snapshot)obj;
+
 			// need to check for nulls because serializer and config snapshots may be null on restore
-			return (obj instanceof Snapshot)
-				&& name.equals(((Snapshot) obj).getName())
-				&& assignmentMode.equals(((Snapshot) obj).getAssignmentMode())
-				&& ((partitionStateSerializer == null && ((Snapshot) obj).getPartitionStateSerializer() == null)
-					|| partitionStateSerializer.equals(((Snapshot) obj).getPartitionStateSerializer()))
-				&& ((partitionStateSerializerConfigSnapshot == null && ((Snapshot) obj).getPartitionStateSerializerConfigSnapshot() == null)
-					|| partitionStateSerializerConfigSnapshot.equals(((Snapshot) obj).getPartitionStateSerializerConfigSnapshot()));
+			return Objects.equals(name, snapshot.getName())
+				&& Objects.equals(assignmentMode, snapshot.getAssignmentMode())
```

Comment:
Yes. You are right. I did this because I want make them have the same way, Objects.equal. Now, I will return to the previously method. Thanks ~